tox.ini

```diff
@@ -210,34 +210,28 @@ commands =
 
 
 [testenv:lint]
-description =
-    lint the Python code with flake8 (code linter), black (code formatter), and
-    isort (sorting of imports)
+description = Run linters and type checks
 package = skip
 deps =
-    flake8
-    flake8-bugbear
-    black
+    ruff
     blackdoc
-    isort
 
 commands =
-    flake8 --extend-exclude "*/build/*" {[testenv]lint_folders}
-    black --check --diff {[testenv]lint_folders}
+    ruff format --diff {[testenv]lint_folders}
+    ruff check {[testenv]lint_folders}
     blackdoc --check --diff {[testenv]lint_folders}
-    isort --check-only --diff --extend-skip-glob "*/build/*" {[testenv]lint_folders}
+
 
 [testenv:format]
 description = Abuse tox to do actual formatting on all files.
 package = skip
 deps =
-    black
+    ruff
     blackdoc
-    isort
 commands =
-    black {[testenv]lint_folders}
     blackdoc {[testenv]lint_folders}
-    isort {[testenv]lint_folders}
+    ruff format {[testenv]lint_folders}
+    ruff check --fix-only {[testenv]lint_folders}
```

Comment:
this auto-fix some of the linting issues?

Reply:
yes it should fix some of the linting issues - basically if you tox format, it should pass tox lint

Reply:
Hmm, should we rename this to `tox -e fix-lint`? Since it does more than formatting? Or is the old name fine?